nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -1410,49 +1461,60 @@ def _run_connected_components(
         deduped_parsed_id_path,
         output_path,
     ):
-        Comms.initialize(p2p=True)
-        df = dask_cudf.read_parquet(
-            deduped_encoded_jaccard_path, blocksize="1GB", aggregate_files=True
-        )
-        df = df[df["jaccard"] == 1].reset_index(drop=True)
+        t0 = time.time()
+        with performance_report_if(
+            self.profile_dir,
+            f"connected-components-run-{datetime.now().strftime('%Y%m%d_%H%M%S')}.html",
+        ):
+
+            Comms.initialize(p2p=True)
```

Comment:
@ayushdg , should we remove this here ?  (Unrelated to the PR)
```suggestion
            Comms.initialize(p2p=True)
```

Reply:
Let's hold off on this update here. We can open a followup that sets this based on the cluster protocol. `True` for ucx and `False` for tcp

Reply:
Lets discuss this on that separate PR.